### PR TITLE
Using Dirichlet BC of Pressure in Navier-Stokes

### LINKED ISSUE #6585: Using Dirichlet BC of Pressure in Navier-Stokes
We can't impose a pressure inlet, as Dirichlet BC, in Navier-Stokes module.
Simulation is converging to a wrong result, see under:
pressurebc_failed
Here is the expected result of the simulation:
pressurebc

### PR DESCRIPTION
**Pull request by mlesueur**
changed weak formulation of gradP in INSMomentum (that solves the issue)
both formulations are accessible via a boolean flag
2 tests implemented to check the difference of the two formulations

see [discussion](https://groups.google.com/forum/#!topic/moose-users/yWi8bdfZWvU)

closes #6585

### CODE REVIEW

**Comment by permcody** [comment]
Holy Merge Commits! Can you clean up this PR so that we can review your changes?

Please see our http://mooseframework.org/wiki/Contributing/ page for more information.

File: modules/navier_stokes/src/kernels/INSMomentum.C [hunk 1]
```diff
@@ -71,7 +76,10 @@ Real INSMomentum::computeQpResidual()
      _w_vel[_qp]*_grad_u[_qp](2)) * _test[_i][_qp];
 
   // The pressure part, -p (div v)
-  Real pressure_part = -_p[_qp] * _grad_test[_i][_qp](_component);
+  Real pressure_part;
+  if(_integrate_p_by_parts)  pressure_part = -_p[_qp] * _grad_test[_i][_qp](_component);
```
Comment (jwpeterson) [inline modules/navier_stokes/src/kernels/INSMomentum.C]: Add space between `if` and opening parenthesis here.

**Comment by jwpeterson** [comment]
Just to expand on what @permcody said, it looks like you based your commits on master branch, then made a PR into devel branch. Rebasing your commits on devel and force-pushing to this branch should fix that issue.  

The commit messages in 7734d9e and caf24e7 are really similar, so I got a bit confused at first.  Since the second one is adding tests, just make the commit message specifically about that.

I've also made an in-line where the formatting problem below occurs.
```
ERROR: The following files contain control keywords without proper spacing (for, if, while, or switch)
http://mooseframework.org/wiki/CodeStandards:
	./modules/navier_stokes/src/kernels/INSMomentum.C
```

**Comment by andrsd** [comment]
@mlesueur 1) You want to develop in a feature branch that branches off of devel. To fix it, try `git rebase --onto devel HEAD~2 HEAD` (this assumes you did only 2 commits (which I saw on your PR). 2) Then, `git checkout -b descriptive-branch-name`. Then push that branch into your fork and open a new PR for it.  I am afraid force-pushing into this PR will not work, since you tried to merge your master branch into our devel.

Also make sure your devel is up to date before you start this.

**Comment by andrsd** [comment]
Also, when you open your PR you should see only your work, not 250+ merge commits. That is a sanity check that you are doing the right thing...

**Comment by mlesueur** [comment]
Hi guys,
Sorry about that. It was my first pull request, it didn't go so well apparently ^^
You know there is a problem when you see 6 new emails in the morning ahahah
I hope I didn't cause any trouble.

I spend the full day trying to get something that works, you should be able to see it on this [pull request](https://github.com/idaholab/moose/pull/6595)
I hope I did things right this time.
Cheers,

Martin

**Comment by jwpeterson** [comment]
Replaced by #6595

**Pull request closed without merging**